security/pkg/nodeagent/caClient/client.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ca
+
+// Client interface defines the clients need to implement to talk to CA for CSR.
+type Client interface {
+	CSRSign(csrPEM []byte /*PEM-encoded certificate request*/, subjectID string, certValidTTLInSec int64) ([]byte /*PEM-encoded certificate chain*/, error)
```

Comment:
Please make sure this API to be consistent with the managed CA one. (i remember justin's shared with us). Otherwise,
/lgtm

Reply:
Should be fine for now. I'll be working on an API here this week and next.